src/app/core/connection.ts

```diff
@@ -634,8 +632,8 @@ export default class Connection {
      * @memberof Connection
      * @returns {void}
      */
-    addWayPoints(wayPoints) {
-        this.__paths.push(wayPoints);
+    addWayPoints(wayPoints: Point) {
```

Comment:
Should be an array of `Point`?

Reply:
That depends on whether `__paths` is meant to be a point array of segment array